app/views/layouts/application.html.erb

```diff
@@ -23,8 +24,15 @@
         <%= link_to "Sign In", new_user_session_path %>
       <% end %>
     </nav>
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
+    <% flash.each do |type, msg| %>
+      <% if msg.kind_of?(Hash) %>
+        <% msg.each do |_, m| %>
+          <%= content_tag(:div, m.join(''), class: "alert alert-#{type}") %>
+        <% end %>
+      <% elsif msg.kind_of?(String) %>
+        <%= content_tag(:div, msg, class: "alert alert-#{type}") %>
+      <% end %>
+    <% end %>
```

Comment:
if you use `resource.errors.full_messages.to_sentence`, then a String is always returned and you won't need to worry about handling a Hash. but this is cool too if preferable.

Reply:
![screen shot 2018-02-06 at 3 19 52 pm](https://user-images.githubusercontent.com/30031865/35882180-4c4ac968-0b51-11e8-8ffa-044f2ad5eb60.png)

^^Just cuz its fun to add screenshots.... When I take out the code related to pulling messages from a Hash, I see the whole errors object: {:url=>["can't be blank"], :base=>[], :title=>[], :text=>[]}

I'm taking out the Hash vs String syntax here.  That was the solution I implemented in grade book after seeing a similar error message in that app.

I’m working on a set up where field-specific messages will show up inline, just below the missing element. If the error is on base, it should show up at just above the top of the form.

Reply:
> When I take out the code related to pulling messages from a Hash, I see the whole errors object

Right, because it's calling `to_s` on `resource.errors`.  If you nab `full_messages` off of that, you should see an array of messages like `["Url can't be blank", "Title can't be blank"]`.  Then, add `to_sentence` to that and get a string like `"Url can't be blank and Title can't be blank"`. So the full code is `resource.errors.full_messages.to_sentence`.  Replacing `resource` with... `submission`(?) in this instance.

I like the idea of the inline errors though 👍